### PR TITLE
Exiles on 1.1

### PR DESCRIPTION
**Pull request by anselor**
Aggregation of many PRs

Integrates the following PRs

#123 
#136 
#138 
#145 
#148 
#150 
#155 

#157 

Supersedes #139

### CODE REVIEW

File: static/ankersrv.js [hunk 1]
```diff
@@ -355,18 +426,31 @@ $(function () {
...
     sockets.ctrl = new AutoWebSocket({
         name: "Control socket",
-        url: `ws://${location.host}/ws/ctrl`,
+        url: `${location.protocol.replace('http','ws')}${location.host}/ws/ctrl`,
         badge: "#badge-ctrl",
     });
 
+    sockets.pppp_state = new AutoWebSocket({
+        name: "PPPP socket",
+        url: `ws://${location.host}/ws/pppp-state`,
```
Comment (sondregronas) [inline static/ankersrv.js]: Found an error in the pppp_state that would prevent PPPP connection over https:

```suggestion
        url: `${location.protocol.replace('http','ws')}${location.host}/ws/pppp-state`,
```

Comment (anselor) [reply static/ankersrv.js]: @treitmayr This was from your changes in this PR:
https://github.com/Ankermgmt/ankermake-m5-protocol/pull/145/commits/1089b493d13e6c3b6abbe4a9168972407702e8c4

Comment (treitmayr) [reply static/ankersrv.js]: My changes were done in a consistent way in the existing code base at the time of the commit. However, the merge of [d6c727b](https://github.com/Ankermgmt/ankermake-m5-protocol/pull/160/commits/d6c727b378140d58395d50f12482fec53864b1d3) introduced wss support, again on the code base from back then. This is the explanation for the current state.
However, I agree with @sondregronas to change the affected line in the proposed way.

**Comment by sondregronas** [comment]
Just commenting here to say that I've opened a PR to this PR with some more changes - let me know if you want me to switch targets to the main repo. PR: https://github.com/anselor/ankermake-m5-protocol/pull/4

### Changes within that PR to this one

- updating `pycryptodomex`, `flask`, `flask-sock` to the latest version + adding `python-dotenv` to dependencies
- addresses the change mentioned above in `ankersrv.js`
- restart the pppp api worker (by calling `worker_start()`, which restarts the api state) on pppp failure (see notes below)
- start the videoqueue service on `/video` if it isn't running, allowing webrtc access via go2rtc (see #162)
- bugfix: allow webserver access with missing credentials (webserver couldn't start without any camera present in cfg)
- `/api/ankerctl/status` endpoint, displays a JSON of every services state and the `app_api_version` dict

### Note regarding pppp:

With this version, pppp should not crash (at least not as often). I've noticed before it would stop working after 2-3 hours of uptime, (but that could also be my really old xeon failing). Instead of crashing it will "flicker" for lack of better words, where sending gcode will _sometimes_ work, and the videostream will have corrupted B/P-frames that only get worse over time. Reloading services are to no avail. Restarting the entire application every 2 hours does circumvent this as far as I can tell.

I tried getting the webserver to run with a different WSGI server, but had no luck. I only tested `gevent` or `eventlet`. I wanted to try gunicorn but from my understanding it would require some refactoring to how the cfg is loaded so I stopped trying.

### All of these prs combined _should_ resolve (at least to my limited understanding):

#163, #161, #158, partially #156 (PPPP will _work_ but stops being reliable), #154, #151, #147, #127 (might be unrelated?), #112

Might be missing a few issues as well that I haven't encountered myself.

**Pull request closed without merging**